### PR TITLE
Add desktop Debug provider for App Check

### PR DESCRIPTION
**Pull request by a-maurice**
Add the desktop implementation of the debug provider for App Check, which uses the desktop rest implementation to retrieve a valid attestation token from the server.
***

Local testing
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [x] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: app_check/src/desktop/debug_provider_desktop.cc [hunk 1]
```diff
@@ -14,19 +14,118 @@
...
+    AppCheckToken token;
+    token.token = std::move(response.token());
+    // Expected response is in seconds
+    int64_t extra_time = strtol(response.ttl().c_str(), nullptr, 10);
+    token.expire_time_millis = (response.fetch_time() + extra_time) * 1000;
+    completion_callback(token, kAppCheckErrorNone, "");
+  } else {
+    // Create an error message, and pass it along instead.
+    AppCheckToken token;
+    char error_message[1000];
```
Comment (AlmostMatt) [inline app_check/src/desktop/debug_provider_desktop.cc:79]: It seems a bit strange to me that we are working with a simple character array here when the callback wants a string, but this does seem to truncate the response body to at most 1000 characters which is nice. 
(No action required)

Comment (a-maurice) [reply app_check/src/desktop/debug_provider_desktop.cc:79]: Yeah, seemed like the safest way to handle it.

File: app_check/src/desktop/debug_provider_desktop.cc [hunk 2]
```diff
@@ -14,19 +14,118 @@
...
+    snprintf(error_message, sizeof(error_message),
+             "The server responded with an error.\n"
+             "HTTP status code: %d \n"
+             "Response body: %s\n",
+             response.status(), response.GetBody());
+    completion_callback(token, kAppCheckErrorServerUnreachable, error_message);
```
[the remark this answers is not in the document]
Comment (a-maurice) [reply app_check/src/desktop/debug_provider_desktop.cc:85]: Sounds good.

File: app_check/src/desktop/debug_token_request.h [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      : RequestJson(debug_token_request_resource_data) {
+    std::string server_url(
+        "https://firebaseappcheck.googleapis.com/v1beta/projects/");
+    server_url.append(app->options().project_id());
+    server_url.append("/apps/");
+    server_url.append(app->options().app_id());
+    server_url.append(":exchangeDebugToken");
+    set_url(server_url.c_str());
+
+    add_header("X-Goog-Api-Key", app->options().api_key());
```
[the remark this answers is not in the document]
Comment (a-maurice) [reply app_check/src/desktop/debug_token_request.h:46]: Yeah. I moved it out to a constant as well, with a comment, but it is basically a one off that exists for this.

**Review by AlmostMatt — APPROVED** [review]

**Pull request merged**